addons/web/static/src/js/core/utils.js

```diff
@@ -303,6 +303,40 @@ var utils = {
             reader.readAsDataURL(file);
         });
     },
+    /**
+     * Returns an object holding different groups defined by a given criterion
+     * or a default one. Each group is a subset of the original given list.
+     * The given criterion can either be:
+     * - a string: a property name on the list elements which value will be the
+     * group name,
+     * - a function: a handler that will return the group name from a given
+     * element.
+     *
+     * @param {any[]} list
+     * @param {string | (element: any) => string} criterion
+     * @returns {Object}
+     */
+    groupBy: function (list, criterion) {
+        let getter;
+        if (criterion) {
+            switch (typeof criterion) {
+                case 'string': getter = element => element[criterion]; break;
+                case 'function': getter = criterion; break;
+                default: throw new Error(`Expected criterion of type 'string' or 'function' and got '${typeof criterion}'`);
+            }
+        } else {
+            getter = element => element;
```

Comment:
in case this getter is used and elements passed to it are not strings, they will be stringified implicitely --> better use JSON.stringify?
getter = element => JSON.stringify(element)?

Reply:
I actually tried to get as close as possible to `_.groupBy` so we can easily replace the underscore by `utils` wherever it's used. Underscore.js seems to cast the group names as string (implicitly or not) while stringifying strings will surround the value with double quotes.

I can however explicitly cast the group names as string if it is easier and more consistent to read.

Reply:
But again String(false) === String("false") for example

Reply:
Indeed, but so does `_.groupBy`. At any rate I think it is not recommended to return anything else than a string (though it is supported).

I think what would be interesting to discuss is whether  this (and other methods like `sortBy`) should behave *exactly* like underscore does. If it is not the case it would then be interesting to stringify values, or even better throw an error if anything else than a string is returned.

@aab-odoo mind to give us an input on this subject?

Reply:
Let us add a check, that a string is returned, no?